presto-main/src/main/java/io/prestosql/sql/planner/LiteralEncoder.java

```diff
@@ -224,6 +237,22 @@ public Expression toExpression(Object object, Type type)
             return new GenericLiteral("DATE", new SqlDate(toIntExact((Long) object)).toString());
         }
 
+        if (type.equals(LIKE_PATTERN) || type.equals(JONI_REGEXP)) {
```

Comment:
Maybe we should add an interface with `Expression toLiteral()`?

Reply:
To Type? I don’t think that’s a good idea. Types shouldn’t have a dependency on the AST/IR.